src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -1171,6 +1169,27 @@ protected void _renameUsing(Map<String, POJOPropertyBuilder> propMap,
         }
     }
 
+    /**
+     * Helper method called to check if given property should be renamed using {@link PropertyNamingStrategies}
+     *
+     * @since 2.16
+     */
+    private boolean _shouldRename(POJOPropertyBuilder prop) {
+        // [databind#4409]: Need to skip renaming for Enums, unless OBJECT format
+        if (getType().isEnumType() && Objects.equals(prop.getPrimaryType(), getType())) {
+            if (prop.isObjectFormatShape()) {
```

Comment:
This is in wrong place: it would check for annotation of a property of Enum type referring to another type and being annotated as `Shape.OBJECT` -- that would not cover cases where `Enum` class is annotated or -- if supported, not sure it is -- property of another (almost certainly non-Enum) type pointing Enum type and having Shape override.